optimum/commands/export/ipex.py

```diff
@@ -0,0 +1,118 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Defines the command line for the export with IPEX."""
+
+import logging
+import sys
+from pathlib import Path
+from typing import TYPE_CHECKING, Optional
+
+from huggingface_hub.constants import HUGGINGFACE_HUB_CACHE
+
+from ...exporters import TasksManager
+from ..base import BaseOptimumCLICommand, CommandInfo
+
+
+logger = logging.getLogger(__name__)
+
+
+if TYPE_CHECKING:
+    from argparse import ArgumentParser, Namespace, _SubParsersAction
+
+
+def parse_args_ipex(parser: "ArgumentParser"):
+    required_group = parser.add_argument_group("Required arguments")
+    required_group.add_argument(
+        "-m", "--model", type=str, required=True, help="Model ID on huggingface.co or path on disk to load model from."
+    )
+    required_group.add_argument(
+        "output", type=Path, help="Path indicating the directory where to store the generated IPEX model."
+    )
+    optional_group = parser.add_argument_group("Optional arguments")
+    optional_group.add_argument(
+        "--task",
+        default="auto",
+        help=(
+            "The task to export the model for. If not specified, the task will be auto-inferred based on the model. Available tasks depend on the model."
+        ),
+    )
+    optional_group.add_argument(
+        "--trust_remote_code",
+        action="store_true",
+        help=(
+            "Allows to use custom code for the modeling hosted in the model repository. This option should only be set for repositories you trust and in which "
+            "you have read the code, as it will execute on your local machine arbitrary code present in the model repository."
+        ),
+    )
+    optional_group.add_argument("--revision", default=None, help="model kwargs")
+    optional_group.add_argument("--token", default=None, help="model kwargs")
+    optional_group.add_argument("--cache_dir", type=str, default=HUGGINGFACE_HUB_CACHE, help="model kwargs")
+    optional_group.add_argument("--subfolder", type=str, default="", help="model kwargs")
+    optional_group.add_argument("--local_files_only", type=bool, default=False, help="model kwargs")
+    optional_group.add_argument("--force_download", type=bool, default=False, help="model kwargs")
+    optional_group.add_argument("--commit_hash", default=None, help="model kwargs")
+    optional_group.add_argument("--torch_dtype", type=str, default="float32", help="model kwargs")
+
+
+class IPEXExportCommand(BaseOptimumCLICommand):
```

Comment:
could you add test for this ?

Reply:
it's not realted to documentation so I think we can split this PR in two

Reply:
Do you have examples for testing command line?